content/en/docs/usage/certificate.md

```diff
@@ -91,71 +90,159 @@ The `Certificate` will be issued using the issuer named `ca-issuer` in the
+reissuance of the Certificate object (see
+[Actions that will trigger a rotation of the private key](#actions-triggering-private-key-rotation) below).
+
+With `rotationPolicy: Always`, cert-manager waits until the Certificate
+object is correctly signed before overwriting the `tls.key` file in the
+Secret.
 
-When [requesting certificates using ingress-shim](../ingress/), the component
-`ingress-gce`, if used, requires that a temporary certificate is present while
-waiting for issuance of a signed certificate when serving. To facilitate this,
-if the annotation `"cert-manager.io/issue-temporary-certificate": "true"` is
-present on the certificate, a self signed temporary certificate will be present
-on the `Secret` until it is overwritten once the signed certificate has been
-issued.
+With this setting, you can expect **no downtime** if your application can
+detect changes to the mounted `tls.crt` and `tls.key` using and reload them
+gracefully or automatically restart.
 
-## Configuring private key rotation
+If your application only loads the private key and signed certificate once
+at start up, the new certificate won't immediately be served by your
+application, and you will want to either manually restart your pod with
+`kubectl rollout restart`, or automate the action by running
+[wave](https://github.com/wave-k8s/wave). Wave is a Secret controller that
+makes sure deployments get restarted whenever a mounted Secret changes.
 
-When a certificate is re-issued for any reason, including because it is nearing
-expiry, when a change to the spec is made or a re-issuance is manually
-triggered, cert-manager supports configuring the 'private key rotation policy'
-to either always re-use the existing private key (the default behavior) or to
-regenerate a new private key on each issuance (the recommended behavior).
+{{% alert title="Re-use of private keys" color="primary" %}}
 
-This is configured using the `spec.privateKey.rotationPolicy` like so:
+Some issuers, like the built-in [Venafi
+issuer](/docs/configuration/venafi/), may disallow re-using private keys.
+If this is the case, you must explicitly configure the `rotationPolicy:
+Always` setting for each of your Certificate objects accordingly.
+
+{{% /alert %}}
+
+In the following example, the certificate has been set with
+`rotationPolicy: Always`:
 
 ```yaml
 apiVersion: cert-manager.io/v1
 kind: Certificate
-metadata:
-  name: my-cert
-  ...
 spec:
   secretName: my-cert-tls
-  dnsNames:
-  - example.com
   privateKey:
-    rotationPolicy: Always
+    rotationPolicy: Always # 🔰 Here.
+```
+
+### Actions that will trigger a rotation of the private key {#actions-triggering-private-key-rotation}
+
+Setting the `rotationPolicy: Always` or change the Certificate's spec won't
+rotate the private key immediately. In order to get the private key secret
+rotated, the certificate objects must be reissued. A certificate object is
+reissued with either:
```

Comment:
```suggestion
Setting the `rotationPolicy: Always` or changing other fields in the Certificate spec won't
rotate the private key immediately. In order to rotate the private key, the certificate objects must be reissued. A certificate object is
reissued under the following circumstances:
```

Reply:
Also:

> Changing the Certificate spec does not trigger the renewal process,

This statement in isolation is not true, and may cause confusion. The *majority* of fields *will* cause a re-issuance (e.g. changing dnsNames, commonName, usages etc). Changing the `rotationPolicy` (or any other fields that control how a renewal happens) *won't* cause a re-issuance immediately.

Reply:
In https://github.com/cert-manager/website/pull/521#discussion_r627557993, Richard proposed to remove the line:

> The certificate object is reissued when a change to the certificate object's spec is made.

I think I misunderstood Richard's suggestion back in May. I should have understood Richard's comment as:

> The certificate object is reissued when a change to one of the certificate object's fields `commonName`, `dnsNames`, `ipAddresses`, `uris`, `emailAddresses`, `subject`, `isCA`, `usages`, `duration` or `issuerRef` is made.

(as per [`RequestMatchesSpec`](https://github.com/jetstack/cert-manager/blob/8ae179b8f5aec708f8cb320a40ca8b495c9d889e/pkg/controller/certificates/util.go#L118))

Apologies for my misunderstanding.

I re-added the line about reissuing on spec change in ea1f25b.